bluefog/common/mpi_controller.cc

```diff
@@ -969,6 +975,113 @@ Status MPIController::WinMutexRelease(const std::string& name,
   return MPIWinMutexReleaseImpl(mutex_win, release_ranks, mpi_ctx_.rank_, is_sync);
 }
 
+Status MPIController::WinVersionGetUpdate(const std::string& name,
+                                          const std::vector<int>& ranks) {
+  BFLOG(TRACE, mpi_ctx_.rank_) << "Update Win Version for " << name << ".";
+
+  auto it = mpi_ctx_.named_win_map.find(name);
+  if (it == mpi_ctx_.named_win_map.end()) {
+    return Status::PreconditionError(
+        "Cannot accquire Version Win for " + name +
+        ". It may not be created or has "
+        "been destroyed or wrong name for associated window.");
+  }
+  std::shared_ptr<MPI_Win> version_win = it->second->GetVersionWin();
+  if (!version_win) {
+    return Status::PreconditionError("Cannot accuire Version Win for " + name +
```

Comment:
1) acquire

2) "acquire" is typically used with mutex. For version window just call set or get?